### PR TITLE
first pass of istioctl proxy-config workload command

### PR DESCRIPTION
**Pull request by GregHanson**
**Please provide a description of this PR:**
Part of https://github.com/istio/ztunnel/issues/87

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ X ] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ X ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

**Comment by GregHanson** [comment]
sample of CLI output using the ztunnel setup steps from the ambient branch [here](https://github.com/istio/istio/blob/experimental-ambient/CONTRIBUTING.md):
```
$ CGO_ENABLED=0 go run istioctl/cmd/istioctl/main.go proxy-config workload ztunnel-g6xmf.istio-system
WORKLOAD IP NODE                  NAME                   PROTOCOL
10.244.0.2  ambient-control-plane coredns                Tcp
10.244.0.2  ambient-control-plane coredns                Tcp
10.244.0.3  ambient-control-plane local-path-provisioner Tcp
10.244.0.3  ambient-control-plane local-path-provisioner Tcp
10.244.0.4  ambient-control-plane coredns                Tcp
10.244.0.4  ambient-control-plane coredns                Tcp
10.244.0.5  ambient-control-plane ztunnel                Tcp
10.244.0.5  ambient-control-plane ztunnel                Tcp
10.244.1.2  ambient-worker2       istiod                 Tcp
10.244.1.2  ambient-worker2       istiod                 Tcp
10.244.1.3  ambient-worker2       ztunnel                Tcp
10.244.1.3  ambient-worker2       ztunnel                Tcp
10.244.1.4  ambient-worker2       helloworld             Hbone
10.244.1.4  ambient-worker2       helloworld             Hbone
10.244.1.5  ambient-worker2       sleep                  Hbone
10.244.1.5  ambient-worker2       sleep                  Hbone
10.244.2.2  ambient-worker        ztunnel                Tcp
10.244.2.2  ambient-worker        ztunnel                Tcp
10.244.2.3  ambient-worker        istio-ingressgateway   Tcp
10.244.2.3  ambient-worker        istio-ingressgateway   Tcp
10.244.2.7  ambient-worker        helloworld             Hbone
10.244.2.7  ambient-worker        helloworld             Hbone
10.244.2.8  ambient-worker        swiss-army-knife       Hbone
10.244.2.8  ambient-worker        swiss-army-knife       Hbone
10.244.2.9  ambient-worker        sleep2                 Tcp
10.244.2.9  ambient-worker        sleep2                 Tcp
```

File: istioctl/pkg/util/configdump/workload.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package configdump
+
+type ZtunnelWorkload struct {
+	WorkloadIP        string   `json:"workload_ip"`
```
Comment (howardjohn) [inline istioctl/pkg/util/configdump/workload.go:18]: we should switch ztunnel to camelCase to match rest of Istio probably

**Review by howardjohn — COMMENTED** [review]
mostly lgtm, I am getting 2x each entry though:

```
WORKLOAD IP NODE                  NAME                   PROTOCOL
10.244.0.2  ambient-control-plane local-path-provisioner Tcp
10.244.0.2  ambient-control-plane local-path-provisioner Tcp
10.244.0.3  ambient-control-plane coredns                Tcp
10.244.0.3  ambient-control-plane coredns                Tcp
10.244.0.4  ambient-control-plane coredns                Tcp
10.244.0.4  ambient-control-plane coredns                Tcp
10.244.0.5  ambient-control-plane metallb                Tcp
10.244.0.5  ambient-control-plane metallb                Tcp
10.244.0.6  ambient-control-plane shell                  Tcp
10.244.0.6  ambient-control-plane shell                  Tcp
10.244.0.7  ambient-control-plane istiod                 Tcp
10.244.0.7  ambient-control-plane istiod                 Tcp
10.244.0.8  ambient-control-plane ztunnel                Tcp
10.244.0.8  ambient-control-plane ztunnel                Tcp
10.244.0.9  ambient-control-plane istio-ingressgateway   Tcp
10.244.0.9  ambient-control-plane istio-ingressgateway   Tcp

```

**Comment by GregHanson** [comment]
@howardjohn @stevenctl anything from the other two fields returned from the [config_dump API ](https://github.com/istio/ztunnel/blob/master/src/workload.rs#L358) worth including?
```
    "workload_to_vip":
    {
        "10.244.1.6":
        [
            [
                "10.96.84.115:5000",
                5000
            ],
            [
                "10.96.32.70:5000",
                5000
            ]
        ],
        ...
    },
    "vips":
    {
        "10.96.32.70:5000":
        [
            {
                "workload":
                {
                    "workload_ip": "10.244.1.6",
                    "waypoint_address": null,
                    "gateway_ip": null,
                    "protocol": "Hbone",
                    "name": "helloworld-v1-cross-node-55446d46d8-fnl6v",
                    "namespace": "default",
                    "service_account": "helloworld",
                    "workload_name": "helloworld-v1-cross-node",
                    "workload_type": "deployment",
                    "canonical_name": "helloworld",
                    "canonical_revision": "v1",
                    "node": "ambient-worker",
                    "native_hbone": false
                },
                "port": 5000
            }
        ],
        ...
    }
```
The only main addition I see is the `port` value

**Comment by howardjohn** [comment]
The API currently is really 2 parts: workloads and services

We could either make these 2 different commands (really probably 3 total, since we will have 'secret' as well) or maybe make it `NAME,NAMESPACE,...,VIPS` or something (where VIPS is a list of IPs, probably no need for port)

File: istioctl/cmd/proxyconfig.go [hunk 2]
```diff
@@ -520,6 +551,79 @@ func allConfigCmd() *cobra.Command {
...
+			if len(args) == 1 {
+				if podName, podNamespace, err = getPodName(args[0]); err != nil {
+					return err
+				}
+				if !strings.Contains(podName, "ztunnel") {
+					return fmt.Errorf("workloads command is only supported by ztunnel proxies: %v", podName)
+				}
+				configWriter, err = setupZtunnelConfigDumpWriter(podName, podNamespace, c.OutOrStdout())
+			} else {
+				//configWriter, err = setupFileConfigdumpWriter(configDumpFile, c.OutOrStdout())
```
[the remark this answers is not in the document]
Comment (GregHanson) [reply istioctl/cmd/proxyconfig.go]: fixed the file reading behavior

File: istioctl/pkg/writer/ztunnel/configdump/workload.go [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+	if filter.Verbose {
+		// TODO
+		fmt.Fprintln(w, "WORKLOAD IP\tNODE\tNAME\tPROTOCOL")
+	} else {
+		fmt.Fprintln(w, "WORKLOAD IP\tNODE\tNAME\tPROTOCOL")
+	}
+	for _, wl := range verifiedWorkloads {
+		if filter.Verbose {
+			// TODO
+			fmt.Fprintf(w, "%v\t%v\t%v\t%v\n", wl.WorkloadIP, wl.Node, wl.CanonicalName, wl.Protocol)
```
[the remark this answers is not in the document]
Comment (GregHanson) [reply istioctl/pkg/writer/ztunnel/configdump/workload.go]: done

**Comment by GregHanson** [comment]
new table view:
```
go run istioctl/cmd/istioctl/main.go proxy-config workload ztunnel-g6xmf.istio-system
NAME                                      NAMESPACE          IP         NODE                  PROTOCOL
coredns-6d4b75cb6d-kt4sq                  kube-system        10.244.0.4 ambient-control-plane Tcp
coredns-6d4b75cb6d-xmg7p                  kube-system        10.244.0.2 ambient-control-plane Tcp
helloworld-v1-cross-node-55446d46d8-4hkl7 default            10.244.2.7 ambient-worker        Hbone
helloworld-v2-same-node-65bd845857-4vzj6  default            10.244.1.4 ambient-worker2       Hbone
istio-ingressgateway-8494dc7655-w7fgm     istio-system       10.244.2.3 ambient-worker        Tcp
istiod-747687bbb6-bcmcm                   istio-system       10.244.1.2 ambient-worker2       Tcp
local-path-provisioner-9cd9bd544-czbw5    local-path-storage 10.244.0.3 ambient-control-plane Tcp
sleep-66b946687b-6rntb                    default            10.244.1.5 ambient-worker2       Hbone
sleep2-7886c67bf8-rrm6g                   default            10.244.2.9 ambient-worker        Tcp
swiss-army-knife-58c658bdfb-df8wh         default            10.244.2.8 ambient-worker        Hbone
ztunnel-g6xmf                             istio-system       10.244.1.3 ambient-worker2       Tcp
ztunnel-glmkk                             istio-system       10.244.0.5 ambient-control-plane Tcp
ztunnel-vd5kt                             istio-system       10.244.2.2 ambient-worker        Tcp
```

**Review by howardjohn — APPROVED** [review]

**Pull request merged**